fakebook/src/app/component/like-view/like-view.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { LikeService } from 'src/app/service/like.service';
+
+@Component({
+  selector: 'app-like-view',
+  templateUrl: './like-view.component.html',
+  styleUrls: ['./like-view.component.css']
+})
+export class LikeViewComponent implements OnInit {
+  @Input() count!: number;
+  @Input() postId!: number;
+  @Input() liked!: boolean;
+
+  constructor(private likeService: LikeService) { }
+
+  ngOnInit(): void {
+  }
+
+  submit(liked: boolean): void {
```

Comment:
there's no point in passing in the `liked` value that's already here in `this.liked`. as it is the code is just more confusing and looks like the logic is wrong at a glance.

Reply:
Removed the parameter and just had it be an event that upon click will check the current state of the like variable